File: src/main/java/recipeapplication/repository/PasswordTokenRepository.java
package recipeapplication.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import recipeapplication.model.PasswordResetToken;
import recipeapplication.model.User;

import javax.transaction.Transactional;
import java.util.Optional;

public interface PasswordTokenRepository extends JpaRepository<PasswordResetToken, Long>
{
    Optional<PasswordResetToken> findByToken(String token);

    @Transactional
    void deleteByUser(User user);
}

File: src/main/kotlin/recipeapplication/repository/PasswordTokenRepository.kt
package recipeapplication.repository

import org.springframework.data.jpa.repository.JpaRepository
import recipeapplication.model.PasswordResetToken
import recipeapplication.model.User
import java.util.*
import javax.transaction.Transactional

interface PasswordTokenRepository : JpaRepository<PasswordResetToken, Long> {
    fun findByToken(token: String): Optional<PasswordResetToken>

    @Transactional
    fun deleteByUser(user: User)
}